File: back-end/src/main/java/pl/straburzynski/campfiresongs/session/exception/NotAuthorizedException.java
package pl.straburzynski.campfiresongs.session.exception;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ResponseStatus;
import pl.straburzynski.campfiresongs.exception.CustomException;
import pl.straburzynski.campfiresongs.exception.ErrorResponse;

@ResponseStatus(HttpStatus.FORBIDDEN)
public class NotAuthorizedException extends CustomException {

    public NotAuthorizedException() {
        super(ErrorResponse.builder()
                .message("Not authorized to session")
                .translationKey("exception.not_authorized_to_session")
                .build());
    }
}

File: back-end-kotlin/src/main/kotlin/pl/straburzynski/campfiresongs/session/exception/NotAuthorizedException.kt
package pl.straburzynski.campfiresongs.session.exception

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseStatus
import pl.straburzynski.campfiresongs.exception.CustomException
import pl.straburzynski.campfiresongs.exception.ErrorResponse


@ResponseStatus(HttpStatus.FORBIDDEN)
class NotAuthorizedException : CustomException(
    ErrorResponse(
        emptyMap(),
        "exception.not_authorized_to_session",
        "Not authorized to session"
    )
)
